### ISSUE TITLE
separate_stains output issue

### ISSUE THREAD

**Issue by asmagen** [scikit-image/scikit-image#4629]
I am using separate_stains as follows on an RGB histopathology image and the results aren't sensical because all of the values in each channel are negative.
`ihc_hax = separate_stains(rgb_original, hax_from_rgb)`

See the following input/suggestions in regards to the possible issues with the function [here](https://stackoverflow.com/questions/61092050/identifying-positive-pixels-after-color-deconvolution-ignoring-boundaries?noredirect=1#comment108080962_61092050) for the following comment given to me:
> Compare the implementation in skimage with what is described in the original paper. You'll see several errors in the implementation, most importantly the lack of a division by the max intensity. They should have used -np.log10(rgb/255) (assuming that 255 is the illumination intensity), rater than -np.log10(rgb). – Cris Luengo Apr 20 at 22:47

and the following comment from [here](https://stackoverflow.com/questions/61297083/separate-stains-output-of-one-channel-and-conversion-to-grayscale/61298083?noredirect=1#comment108453264_61298083) in regards to my observation of only negative values:
>  That makes no sense. I don’t have experience with the functions you’re using here, so cannot verify their correctness. Negative values should be fairly rare in the stain unmixing process if stain definitions used are good approximations. – Cris Luengo Apr 19 at 14:08

Reproducible example:
```
>>> import numpy as np
>>> from skimage.color import hax_from_rgb
>>> import skimage
>>> image = skimage.io.imread(work_dir+'/Figure_1.png')
>>> ihc_hax = separate_stains(image, hax_from_rgb)
>>> aec = ihc_hax[:, :, 1]
>>> np.min(aec)
-0.42166126533003073
>>> np.max(aec)
-0.26432206141264536
```

How can we address and resolve this issue?

![Figure_1](https://user-images.githubusercontent.com/11299568/80645415-134ffe80-8a39-11ea-91b8-256e2f32c17e.png)

**Comment by asmagen** [comment]
Any thoughts @alexdesiqueira?

**Comment by alexdesiqueira** [comment]
Hey @asmagen, thank you for your example and clarifications!
I will check our algorithm as soon as possible, based on that info. Let me look up the paper as well, and I'll get back to you with what I found, ok?
Thanks again!

**Comment by asmagen** [comment]
Yes, thank you. I hope to make progress with it ASAP. There's a link to the paper in the original comment (follow the link to stack exchange).

While we're at it, I'm having a small difficulty that you might know how to resolve in the step using the separated stain channels -- I'm trying to subtract the third channel (which is supposed to correspond to green?) from the second channel, which is the AEC (red stain) that I'm looking for, because the third channel captures the background artifacts such as tissue folds (see details and background [here](https://stackoverflow.com/a/61293526/633263)). The problem is that the range of values in that abc-background matrix is -0.1323706675998537 to -0.03260491447491476, which I have no way to interpret and threshold to identify the positive pixels. I tried using img_as_ubyte or rescale_intensity but it didn't help. Can you advise how to I get that image to be in a normal gray scale values of say 0-255 such that I can define a threshold and mask the image to get the positive pixels only?

Thanks

**Comment by asmagen** [comment]
Any news? @alexdesiqueira @sciunto

**Comment by asmagen** [comment]
Any anticipation for when I can work with this function? @emmanuelle

**Comment by alexdesiqueira** [comment]
@asmagen we're working on it. It will take a while, though. In the meantime, please feel free to submit a pull request fixing it :)

**Comment by asmagen** [comment]
How long do you think it’ll take?

**Comment by alexdesiqueira** [comment]
@asmagen apart my current work, I need to:
- Read and understand the paper — from a research area that I still don't know;
- Check the current code, how it behaves right now, and how it should behave;
- Find a solution, write and test the new code;
-  Write tests to ensure that the function will perform as desired.

After that, my core fellows will:
- Check if my solution is reliable;
- Check if my new code won't break another part of the library.

Then, the code will be ready for you to use. Thank you for your patience until there.
That said: please, if you know how to solve the situation, I encourage you to submit a PR with a fix!

**Comment by alexdesiqueira** [comment]
Closing; solved in #4725. Thank you @asmagen!
